docs/wiki/EL/evm.md

```diff
@@ -258,23 +279,35 @@ EVM doesn't have a direct equivalent to `MSTORE8` for reading. You must read the
 
 > EVM memory is shown as blocks of 32 bytes to illustrate how memory expansion works. In reality, it is a seamless sequence of bytes, without any inherent divisions or blocks.
 
+## Calldata
+The **calldata** memory type is very similar to **Memory** as discussed above in that they both store dynamically sized data that is removed after contract execution. However, the **calldata** memory structure stores read-only data originating from a function's parameters.  The EVM will store the parameter as calldata since it's cheaper than copying the value to **Memory**.
+
+###  Reading from calldata
+If the transaction calling this smart contract passes in calldata, the inputted data can be loaded using the `CALLDATALOAD` opcode, which reads 32 bytes from calldata at a given offset and pushes it onto the stack once the program counter reaches its position in the bytecode. More info on the `CALLDATALOAD` opcode can be found [here](https://veridelisi.medium.com/learn-evm-opcodes-v-a59dc7cbf9c9).
+
+
+Additional opcodes for **calldata** include:
+- `CALLDATASIZE`: Puts a number onto the stack, where that number is the number of bytes in calldata.
+- `CALLDATACOPY`: Copies a segment of calldata to memory and takes three stack arguments: destination memory offset, calldata offset, and length in bytes.
+
+
 ## Storage
 
-Storage is designed as a **word-addressed word array**. Unlike memory, storage is associated with an Ethereum account and is **persisted** across transactions as part of the world state.
+Storage is designed as a **word-addressed word array**. Unlike memory, storage is associated with an Ethereum account and is **persisted** across transactions as part of the world state.  It can be thought of as the **database** associated with the smart contract, which is why it contains the contract's "state" variables. Storage size is fixed at 2^256 slots, 32 bytes each.
```

Comment:
```suggestion
Storage is designed as a **word-addressed word array**. Unlike memory, storage is associated with an Ethereum account and is **persisted** across transactions as part of the world state.  It can be thought of as a   key-value **database** associated with the smart contract, which is why it contains the contract's "state" variables. Storage size is fixed at 2^256 slots, 32 bytes each.
```